libraries/Microsoft.Bot.Builder/AppInsightsLoggerMiddleware.cs

```diff
@@ -0,0 +1,151 @@
+        /// <param name="logOriginalMessage"> (Optional) Enable/Disable logging original message name within Application Insights.</param>
+        /// <param name="config"> (Optional) TelemetryConfiguration to use for Application Insights.</param>
+        public AppInsightsLoggerMiddleware(string instrumentationKey, bool logUserName = false, bool logOriginalMessage = false, TelemetryConfiguration config = null)
+        {
+            if (instrumentationKey == null)
+            {
+                throw new ArgumentNullException(nameof(instrumentationKey));
+            }
+
+            var telemetryConfiguration = config ?? new TelemetryConfiguration(instrumentationKey);
+            _telemetryClient = new TelemetryClient(telemetryConfiguration);
+            _logUserName = logUserName;
+            _logOriginalMessage = logOriginalMessage;
+        }
+
+        /// <summary>
+        /// Records incoming and outgoing activities to the Application Insights store.
+        /// </summary>
+        /// <param name="context">The context object for this turn.</param>
+        /// <param name="nextTurn">The delegate to call to continue the bot middleware pipeline.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        /// <seealso cref="ITurnContext"/>
+        /// <seealso cref="Bot.Schema.IActivity"/>
+        public async Task OnTurnAsync(ITurnContext context, NextDelegate nextTurn, CancellationToken cancellationToken)
+        {
+
+            context.Services.Add(AppInsightServiceKey, _telemetryClient);
+
+            // log incoming activity at beginning of turn
+            if (context.Activity != null)
+            {
+                if (string.IsNullOrEmpty((string)context.Activity.From.Properties["role"]))
+                {
+                    context.Activity.From.Properties["role"] = "user";
+                }
+
+                // Context properties for App Insights
+                if (!string.IsNullOrEmpty(context.Activity.Conversation.Id))
+                {
+                    _telemetryClient.Context.Session.Id = context.Activity.Conversation.Id;
+                }
+
+                if (!string.IsNullOrEmpty(context.Activity.From.Id))
+                {
+                    _telemetryClient.Context.User.Id = context.Activity.From.Id;
+                }
+
+                // Log the Application Insights Bot Message Received
+                try
+                {
+                    _telemetryClient.TrackEvent(BotMsgEvent, FillEventProperties(context));
+                }
+                catch (Exception ex)
+                {
+                    _telemetryClient.TrackException(ex);
+                }
+            }
+
+            // process bot logic
```

Comment:
FWIW, the bot isn't necessarily the next thing in the pipeline so this comment is exactly accurate... probably better off just removing.

Reply:
removed